smpplib/client.py

```diff
@@ -170,22 +170,41 @@ def unbind(self):
         except socket.timeout:
             raise exceptions.ConnectionError()
 
-    def send_pdu(self, p):
+    def manage_outstanding_operations(self, sending, pdu):
+        if not self.max_outstanding_operations:
+            return
+
+        if pdu.is_response():
+            operation = '{}{}'.format('S' if sending else 'C', pdu.sequence)
+            try:
+                self.outstanding_operations.remove(operation)
+                self.logger.debug('Unregistering outstanding operation {}'.format(operation))
+            except KeyError:
+                self.logger.warning('Failed to unregistered outstanding operation {}'.format(operation))
+        else:
+            operation = '{}{}'.format('C' if sending else 'S', pdu.sequence)
```

Comment:
Duplicated line which is used in both branches. Could you move it right above the `if`?